contracts/AuraLiquidityMigrator.sol

```diff
@@ -0,0 +1,264 @@
+     * @param symbol The symbol of the new pool
+     * @param tokens The underlying tokens of the balancer pool, have to be sorted as expected by balancer vault.
+     * @param swapFeePercentage The swap fee percentage
+     * @param oracleEnabled Indcates if the pool should be enabled to be an oracle source.
+     * @param owner The owner of the pool
+     * @param minOut The min amount of bpt
+     */
+    function migrateUniswapV2AndCreatePool(
+        address fromLpToken, //Uniswap LP token
+        string memory name,
+        string memory symbol,
+        IERC20[] memory tokens,
+        uint256 swapFeePercentage,
+        bool oracleEnabled,
+        address owner,
+        uint256 minOut
+    ) public returns (address pool, address gauge) {
+        require(tokens.length == 2, "only token pairs");
+        // 1. Remove liquidity
+        (address token0, uint256 amount0, address token1, uint256 amount1) = _removeLiquidityUniswapV2(fromLpToken);
+        uint256[] memory maxAmountsIn = new uint256[](2);
+
+        // sort amounts
+        maxAmountsIn[0] = address(tokens[0]) == token0 ? amount0 : amount1;
+        maxAmountsIn[1] = address(tokens[1]) == token1 ? amount1 : amount0;
+
+        // 2. Create balancer pool
+        pool = _createBalancerPool(name, symbol, tokens, swapFeePercentage, oracleEnabled, owner);
+
+        // 3. Deposit to balancer pool
+        _addLiquidityBalancer(pool, tokens, maxAmountsIn, msg.sender, minOut);
+
+        gauge = bGaugeFactory.create(pool);
+
+        emit PoolCreated(pool, gauge);
+    }
+
+    /**
+     * @dev Migrates a liquidity position, adds liquidity to the pool, and stake the BPT
+     *
+     * @param fromLpToken The LP token to migrate
+     * @param minOut The min amount of bpt
+     * @param tokens The underlying tokens of the balancer pool, have to be sorted as expected by balancer vault.
+     * @param pool The pool address to add liquidity
+     * @param rewardPool The aura reward pool address
+     */
+    function migrateUniswapV2AndJoinPool(
+        address fromLpToken,
+        uint256 minOut,
+        IERC20[] memory tokens,
+        address pool,
+        address rewardPool
+    ) public {
+        require(tokens.length == 2, "only token pairs");
+        // 1. Remove liquidity
+        (address token0, uint256 amount0, address token1, uint256 amount1) = _removeLiquidityUniswapV2(fromLpToken);
+        // sort amounts
+        uint256[] memory maxAmountsIn = new uint256[](2);
+        maxAmountsIn[0] = address(tokens[0]) == token0 ? amount0 : amount1;
+        maxAmountsIn[1] = address(tokens[1]) == token1 ? amount1 : amount0;
+        // 2. Deposit to balancer pool
+        _addLiquidityBalancer(pool, tokens, maxAmountsIn, address(this), minOut);
+
+        uint256 minted = IERC20(pool).balanceOf(address(this));
+        require(minted > 0, "!mint");
+
+        // 3. Deposit to reward pool
+        IERC20(pool).safeIncreaseAllowance(rewardPool, minted);
```

Comment:
There may not be a rewardPool, maybe if address==0, then skip this part and just transfer the LP tokens back to msg.sender